sdk/file_sdk/XMPFileFormat/xmp_file_format.cpp

```diff
@@ -1,15 +1,169 @@
 #include "xmp_file_format.h"
 
+#include "xmp_stream.h"
+#include <XMP.incl_cpp>
+#include <XMP.hpp>
+#include <time.h>
+#include <string_utils.h>
+#include <mutex>
+
+using std::string;
+using std::vector;
+
 namespace mip {
 namespace file {
 
+const XMP_StringPtr kMsipNamespace = "http://www.microsoft.com/msip/1.0/";
+const XMP_StringPtr kMsipLabelNamespace = "http://www.microsoft.com/msip/label/1.0/";
+const XMP_StringPtr kMsipLabelExtendedNamespace = "http://www.microsoft.com/msip/label/extended/1.0/";
+const XMP_StringPtr kLabelsArrayName = "labels";
+const XMP_StringPtr kExtendedStructName = "extended";
+
+bool Initialized = false;
+std::mutex InitMutex;
+
+static string GetStringByKey(_In_ const SXMPMeta* metadata, _In_ const string &labelsItemPath, _In_ const XMP_StringPtr fieldName, _In_ const XMP_StringPtr alternativeFieldName)
+{
+  std::string path, value;
```

Comment:
coding conventions for defining variables like that?

Reply:
You mean the _In_ ? i remeber adding it in AIP for SDL or fxcop or any other too we used.
I will remove it.

Reply:
defining two variables in one line